Homework done

Azadios opened the pull request:
Мне показалось странным, что area_id принадлежит employer, а не vacancy, но решил, что должно быть как в презентации.
И да, в тексте домашнего задания в 4 пункте просилось среднюю среднюю выплату, а в скобках указана формула для среднего интервала выплат. Сделал для средней выплаты, могу переделать.

Azadios committed:
done

Azadios commented on getData.sql:
```diff
@@ -0,0 +1,66 @@
+-- task 4
+WITH gross AS (
+    SELECT
+        CASE WHEN compensation_gross IS TRUE
+             THEN compensation_to
+             ELSE compensation_to / 0.87
+        END AS compensation_to,
+        CASE WHEN compensation_gross IS TRUE
+             THEN compensation_from
+             ELSE compensation_from / 0.87
+        END AS compensation_from
```
Можно ли было используя уже gross значения прямо здесь создать 3 колонку с average_compensation?
Как я понял, можно либо ещё раз написать эти условия при создании 3 колонки, либо создать ещё одну временную таблицу и ещё раз пройтись по уже gross значениям, но нельзя использовать преобразованные данные этого же SELECT'а.

Azadios commented on getData.sql:
```diff
@@ -0,0 +1,66 @@
+LIMIT 10;
+
+-- task 4
+WITH gross AS (
+    SELECT
+        CASE WHEN compensation_gross IS TRUE
+             THEN compensation_to
+             ELSE compensation_to / 0.87
+        END AS compensation_to,
+        CASE WHEN compensation_gross IS TRUE
+             THEN compensation_from
+             ELSE compensation_from / 0.87
+        END AS compensation_from
+    FROM vacancy
+)
```
Можно ли было используя уже gross значения прямо здесь создать 3 колонку с average_compensation?
Как я понял, можно либо ещё раз написать эти условия при создании 3 колонки, либо создать ещё одну временную таблицу и ещё раз пройтись по уже gross значениям, но нельзя использовать преобразованные данные этого же SELECT'а.

eldmitrieva replied:
Да, можно было бы создать третью колонку. Можно было бы и перевод добавить внутрь avg(), но было бы очень неаккуратно

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,66 @@
+)
+SELECT DISTINCT
+    employer_name AS "Компания"
+FROM response_per_vacancy_ordered
```
Тут не хватает сортировки, сортировка в response_per_vacancy_ordered тут уже не работает

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,66 @@
+        COUNT(*) OVER (PARTITION BY vacancy.vacancy_id) AS response_count
+    FROM response
+    INNER JOIN vacancy ON response.vacancy_id = vacancy.vacancy_id
+    INNER JOIN employer ON vacancy.employer_id = employer.employer_id
```
работодатели, у которых нет либо вакансий, либо откликов, не попадут в топ, надо поправить

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,66 @@
+WITH response_per_vacancy_ordered AS (
+    SELECT DISTINCT
+        employer_name,
+        COUNT(*) OVER (PARTITION BY vacancy.vacancy_id) AS response_count
```
ты тут считаешь количество откликов для каждой вакансии, у тебя так какие-то работодатели будут в выборке несколько раз, такого быть не должно

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,66 @@
+-- task 7
+SELECT DISTINCT
+    area_id AS "Город",
+    MIN(response.created_at - vacancy.created_at) OVER (PARTITION BY area_id) AS "Минимальное время отклика",
```
Неправильно. Если в скрипте заполнения данных поставишь разные created_at, увидишь ошибку

Azadios committed:
fix misspell

Azadios committed:
fixed according to @eldmitrieva review

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,82 @@
+-- task 5
+WITH response_per_employer_vacancy AS (
+    SELECT DISTINCT
+        employer_name,
```
У тебя employer_name при создании не указано как уникальное, поэтому чисто теоретически у тебя могут быть несколько работодателей с одинаковым названием, но разными id, здесь они будут считаться одним работодателем.  А если дублей не будет, то и DISTINCT не нужен

Azadios replied:
Добавил проверку на id.
Хотя, наверное, стоило потребовать уникальность имён, но в заданиях нигде явно не говорилось о таком, решил не менять поведение всей таблицы.

eldmitrieva replied:
Уникальность имен не нужна, иногда бывает, что есть компании с одинаковым названием, особенно если это какой-нибудь маленький магазин у дома.

Azadios committed:
fix task 5

Added employer_id to check if distinct. So that should insure that
no employers with same name but different id will be thown away.

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,83 @@
+        employer_name,
+        MAX(response_count) AS max_response_count
+    FROM response_per_employer_vacancy
+    GROUP BY employer_name
```
здесь тогда тоже должен быть employer_id, иначе ты компании с одним названием считаешь как одну, а может быть и так, что две компании с одинаковым названием набрали близкое максимальное количество откликов, тогда они обе должны быть в результирующей выборке, но будет только одна

Azadios replied:
Верно

eldmitrieva commented on getData.sql:
```diff
@@ -0,0 +1,83 @@
+    INNER JOIN employer ON vacancy.employer_id = employer.employer_id
+    GROUP BY vacancy.vacancy_id, area_id
+)
+SELECT DISTINCT
```
зачем distinct, если ты сгруппировал по area_id?

Azadios replied:
Забыл убрать

Azadios committed:
fixed task 5

Fixed previous fix...
